gensim/downloader.py

```diff
@@ -0,0 +1,359 @@
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+    if tar_file is not None:
+        with open(tar_file, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+
+
+def info(data_=None):
+    """Function for retrieving the list of corpora/models, if data name is not provided. If data name
+    is provided, then it gives detailed information about the data.
+
+    Args:
+        data_(string): Name of the corpus/model.
+
+    Returns:
+        <class 'dict'>: It returns the models/corpora names with detailed information about each.
+    """
+    url = "https://raw.githubusercontent.com/RaRe-Technologies/gensim-data/master/list_with_filename.json"
+    response = urlopen(url)
+    data = response.read().decode("utf-8")
+    data = json.loads(data)
+    if data_ is not None:
+        data_ = get_data_name(data_)
+        corpora = data['corpus']
+        models = data['model']
+        if data_ in corpora:
+            logger.info("%s \n", data['corpus'][data_]["desc"])
+            return data['corpus'][data_]["desc"]
+        elif data_ in models:
+            logger.info("%s \n", data['model'][data_]["desc"])
+            return data['model'][data_]["desc"]
+        else:
+            raise Exception(
+                "Incorrect model/corpus name. Choose the model/corpus from the list "
+                "\n {}".format(json.dumps(data, indent=4)))
+    else:
+        return data
+
+
+def get_checksum(data_, status):
+    """Function for retrieving the checksum of a corpus/model
+
+    Args:
+        data_(string): Name of the corpus/model.
+
+    Returns:
+        string: It returns the checksum for corresponding the corpus/model.
+    """
+    key = "checksum_after_" + status
+    data = info()
+    corpora = data['corpus']
+    models = data['model']
+    if data_ in corpora:
+        return data['corpus'][data_][key]
+    elif data_ in models:
+        return data['model'][data_][key]
+
+
+def _download(data_):
+    """Function for downloading and installed corpus/model depending upon it's current status.
+
+    Args:
+        data_(string): Name of the corpus/model.
+    """
+    url = "https://github.com/RaRe-Technologies/gensim-data/releases/download/{f}/{f}.tar.gz".format(f=data_)
+    data_folder_dir = os.path.join(base_dir, data_)
+    data = info()
+    corpora = data['corpus']
+    models = data['model']
+    if data_ not in corpora and data_ not in models:
+        raise Exception(
+            "Incorect Model/corpus name. Use info() or"
+            " python -m gensim.downloader -c to get a list of models/corpora"
+            " available.")
+    compressed_folder_name = "{f}.tar.gz".format(f=data_)
+    compressed_folder_dir = os.path.join(base_dir, compressed_folder_name)
+    if get_data_status(data_) != "downloaded":
+        if not os.path.exists(data_folder_dir):
```

Comment:
If exists - need to remove all broken files and after it creates all that needed.